SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+angular.module('SECP')
+    .directive('chatList', function () {
+    return {
+        restrict: 'E', //E = element, A = attribute, C = class, M = comment
+        scope: {
+            //@ reads the attribute value, = provides two-way binding, & works with functions
+            contacts: '=',
+            contactSelected: '&contactSelectedFn'
+         },
+        templateUrl: 'directives/chat-list/chat-list.html',
+        link: function ($scope, element, attrs) {
+            $(".chat-list-wrapper").niceScroll({autohidemode:'leave'});
+            $scope.setActive = function(contact) {
+                if($scope.activeContact !== contact) {
+                    $scope.activeContact = contact;
+                    $scope.contactSelected({'contact' : contact});
+                }
+            };
+
+           $scope.$watch('contacts', function(contacts) {
+              if (contacts !== undefined && contacts.length > 0) {
```

Comment:
Might need to check if null if i remember js has undefined and null being different. You Can just use truthy falsey instead

Reply:
@tosinoni this is your intended behaviour right?

Reply:
It is the intended behaviour. 

A property that has no definition is undefined, which also means no value, no type and never been referenced before in the scope. On the other hand, null is where the property exists but it hasn't known what its value is.